### PR TITLE
[skia] Fix build for release-only triplets

### PR DESCRIPTION
**Pull request by Chlumsky**
- #### What does your PR fix?
  Fixes the Skia package build for release-only triplets such as `x64-linux-release`.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Release-only triplets in addition to those that were supported previously. No, I have not.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Probably.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Sure, but what is it supposed to do? It doesn't do anything.

### CODE REVIEW

**Review by FrankXie05 — APPROVED** [review]

File: ports/skia/portfile.cmake [hunk 1]
```diff
@@ -105,11 +105,19 @@ function(replace_skia_dep NAME INCLUDES LIBS_DBG LIBS_REL DEFINITIONS)
     list(TRANSFORM INCLUDES PREPEND "${CURRENT_INSTALLED_DIR}")
     cmake_to_gn_list(_INCLUDES "${INCLUDES}")
 
-    find_libraries(_LIBS_DBG "${LIBS_DBG}" "${CURRENT_INSTALLED_DIR}/debug/lib")
-    cmake_to_gn_list(_LIBS_DBG "${_LIBS_DBG}")
+    if(NOT VCPKG_BUILD_TYPE STREQUAL release)
```
Comment (dg0yt) [inline ports/skia/portfile.cmake:108]: Please just use `if(NOT VCPKG_BUILD_TYPE)`. It indicates that both release and debug are being built.

File: ports/skia/portfile.cmake [hunk 2]
```diff
@@ -105,11 +105,19 @@ function(replace_skia_dep NAME INCLUDES LIBS_DBG LIBS_REL DEFINITIONS)
...
+    endif()
 
-    find_libraries(_LIBS_REL "${LIBS_REL}" "${CURRENT_INSTALLED_DIR}/lib")
-    cmake_to_gn_list(_LIBS_REL "${_LIBS_REL}")
+    if(NOT VCPKG_BUILD_TYPE STREQUAL debug)
+        find_libraries(_LIBS_REL "${LIBS_REL}" "${CURRENT_INSTALLED_DIR}/lib")
+        cmake_to_gn_list(_LIBS_REL "${_LIBS_REL}")
+    else()
+        cmake_to_gn_list(_LIBS_REL "")
+    endif()
```
Comment (dg0yt) [inline ports/skia/portfile.cmake:121]: Unneeded. The variant named "release" is always built.

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by Chlumsky** [comment]
I respectfully disagree with the changes proposed by @dg0yt. My version is both much more clear and more future-proof. Yes, I read in the documentation that `VCPKG_BUILD_TYPE` can be either `release` or empty. But it is *not* called `VCPKG_BUILD_RELEASE_ONLY` (which is a design flaw on your part), so just `if(NOT VCPKG_BUILD_TYPE)` (meaning not release-only) is extremely confusing, and if someone in the future decides that the variable can have other values (nothing is set in stone), my version would still work, but not the proposed one. (Same argument for the claim that release version is always built.)

**Comment by Chlumsky** [comment]
@dg0yt Could this please be merged? It would reduce the size of our Docker images by like 75%...

**Comment by Chlumsky** [comment]
Kind of ironic to put "requires author response" on this even though the author is actually the only one responding...

**Review by FrankXie05 — DISMISSED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**